examples/net_framework/README.md

```diff
@@ -0,0 +1,172 @@
+
+```bash
+mklink /d "D:\cloned\tensorflow\native_client" "D:\cloned\DeepSpeech\native_client"
+```
+
+## Adding environment variables
+
+After you install the requirements there are few environment variables that we need to add.
+
+#### MSYS2 paths
+
+For MSYS2 we need to add `C:\msys64\usr\bin` to the environment variables then we need to run:
+
+```bash
+pacman -Syu
+```
+
+```bash
+pacman -Su
+```
+
+```bash
+pacman -S patch unzip
+```
+
+#### BAZEL path
+
+For BAZEL we need to add the path to the executable, very important to rename the executable to `bazel`.
+
+To check the version installed you can run:
+
+```bash
+bazel version
+```
+
+#### PYTHON path
+
+Add `python.exe` path to the environment variables.
+
+
+#### CUDA paths
+
+If you want to enable CUDA support you need to add the following paths to the environment variables.
+
+```
+C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0
+```
+
+```
+C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\extras\CUPTI\libx64
+```
+
+### Building the code
+
+There's one last command to run before building, you need to run the [configure.py](https://github.com/mozilla/tensorflow/blob/master/configure.py) inside `tensorflow` cloned directory.
+
+Run:
+
+```bash
+python configure.py
+```
+The script will ask the configuration that you want to use, for AVX/AVX2 we will specify manually in the command so, we will skip this option by hitting `enter`. If you want to support CUDA please for this option select `y` when it asks for CUDA support.
+
+Here's one example of the configuration:
+
+```
+Please input the desired Python library path to use.  Default is [D:\py\lib\site-packages]
+
+Do you wish to build TensorFlow with Apache Ignite support? [Y/n]: n
+No Apache Ignite support will be enabled for TensorFlow.
+
+Do you wish to build TensorFlow with XLA JIT support? [y/N]: n
+No XLA JIT support will be enabled for TensorFlow.
+
+Do you wish to build TensorFlow with ROCm support? [y/N]: n
+No ROCm support will be enabled for TensorFlow.
+
+Do you wish to build TensorFlow with CUDA support? [y/N]: n
+No CUDA support will be enabled for TensorFlow.
+
+Please specify optimization flags to use during compilation when bazel option "--config=opt" is specified [Default is /arch:AVX]:
+
+
+Would you like to override eigen strong inline for some C++ compilation to reduce the compilation time? [Y/n]: n
+Not overriding eigen strong inline, some compilations could take more than 20 mins. 
+```
+
+
+### Running the build commands
+
+At this point we are ready to start building the library, go to `tensorflow` directory that you cloned, following our examples should be `D:\cloned\tensorflow`.  
+
+#### CPU
+We will add AVX/AVX2 support in the command, you can remove the flags if you don't want to support them.
+
+```bash
+bazel build -c opt --copt=/arch:AVX --copt=/arch:AVX2 //native_client:libdeepspeech.so
```

Comment:
We document only `-O3` to avoid people doing too many mistakes with flags that don't work on their CPUs, we should likely do the same here.

Reply:
MSVC is ignoring `-O3`, in my  notes I have that `--config=opt` replaces `-O3`, related to https://github.com/tensorflow/tensorflow/issues/7449
Do you prefer to remove `--copt=/arch:AVX` `--copt=/arch:AVX2` and got with `--config=opt` or keep both examples? 

Running test to confirm this.

Reply:
Ok, I was wrong  about `--config=opt` it will not use the instructions that you CPU supports, it will use the instructions that you set in the `configure.py`. I think there's no `-O3` for MSVC.

Reply:
Isn't /O2 the equivalent there? https://docs.microsoft.com/en-us/cpp/build/reference/o-options-optimize-code?view=vs-2017

Reply:
No, `/O2` is enabled by `-c opt` and you still need to specify avx flags.

You can see that `/O2` is still used with `arch:AVX` in the answer https://stackoverflow.com/questions/18371797/qmake-handling-options-for-both-gcc-and-msvc